src/test/java/no/unit/nva/institution/proxy/HttpExecutorImplTest.java

```diff
@@ -178,10 +182,26 @@ public void getNestedInstitutionsReturnsSuccessWhenHttpClientThrowsExceptionInFi
         assertThat(response, is(equalTo(expectedJson)));
     }
 
+    @Test
+    public void getNestedInstitutionsThrowsFailedHttpRequestExceptionWhenHttpClientFailsRepeatedlyOnForSameUri()
+        throws InvalidUriException {
+        HttpClientThrowsExceptionOnNestedInstitutionQueries mockClientContainer =
+            new HttpClientThrowsExceptionOnNestedInstitutionQueries(Language.ENGLISH);
+        HttpClient client = mockClientContainer.getMockClient();
+        HttpExecutorImpl executor = new HttpExecutorImpl(client);
+        Executable action = () -> executor.getNestedInstitution(URI.create(INSTITUTION_REQUEST_URI), Language.ENGLISH);
+
+        FailedHttpRequestException actualException = assertThrows(FailedHttpRequestException.class, action);
+        StringWriter sw = new StringWriter();
```

Comment:
This should be replaced by a nva-testutils library method when nva-testutils is updated